db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java

```diff
@@ -9321,6 +9333,32 @@ else if( routinePermsKey.getGrantee().equals( sd.getAuthorizationId()))
         else if (key instanceof SchemaPermsDescriptor) {
             SchemaPermsDescriptor schemaPermsKey = (SchemaPermsDescriptor) key;
             permissions = getUncachedSchemaPermsDescriptor(schemaPermsKey);
+
+            if( permissions == null)
+            {
+                try {
+                    // The owner has all privileges unless they have been revoked.
+                    SchemaDescriptor sd = getSchemaDescriptor(schemaPermsKey.getSchemaUUID(), ConnectionUtil.getCurrentLCC().getTransactionExecute());
+
+                    if (metadataAccessRestrictionEnabled && sd.isSystemViewSchema() ||
+                            !metadataAccessRestrictionEnabled && sd.isSystemSchema()) {
+                        // allow read access to public
+                        permissions = new SchemaPermsDescriptor(this,
+                                schemaPermsKey.getGrantee(),
+                                (String) null,
+                                schemaPermsKey.getSchemaUUID(),
+                                "Y", "N", "N", "N", "N", "N", "N", "Y");
```

Comment:
What are the `Y`s and `N`s? They definitely beg some more elaborative constants as it's not clear what they stand for.

Reply:
Each argument here represents whether one kind of privilege is granted or not. For example, the first is SELECT privilege.  "Y" means privilege granted with grant option, "y" means privilege granted without grant option, "N" means privilege not granted.

Reply:
Why not to have them be true / false then? If that's too much to ask for, why not to define a constant `ENABLED` and `DISABLED` for `Y` and `N`, respectively?